File: executable/julia/YIGnBu-heatmap.jl
using Plotly
Plotly.signin("test-runner", "9h29fe3l0x")

size = 50
z = rand(size, size)
for r = 1:size
  for c = 1:size
        z(r,c) = sqrt(r*c/size^2)

data = [
  [
    "z" => z, 
    "scl" => "YIGnBu", 
    "type" => "heatmap"
  ]
]
layout = ["title" => "YIGnBu"]

response = Plotly.plot([data], ["layout" => layout, "filename" => "YIGnBu-heatmap", "fileopt" => "overwrite"])
plot_url = response["url"]

File: auto-docs/test_executables/python/YIGnBu-heatmap.py
import plotly.plotly as py
from plotly.graph_objs import *

py.sign_in('theengineear', 'o9zlr0hy6z')

import numpy as np

size = 50
z = np.zeros((size, size)).tolist()
for r in range(1, size + 1):
    for c in range(1, size + 1):
        z[r-1][c-1] = np.sqrt(r*c/float(size**2))
data = Data([
    Heatmap(
        z=z,
        scl='YIGnBu'
    )
])
layout = Layout(
    title='YIGnBu'
)
fig = Figure(data=data, layout=layout)

if 'fig' not in locals():
    if 'data' not in locals():
        raise Exception('no data OR figure!!')
    fig = dict(data=data)  # assumes fig or data
if 'layout' not in fig:
    fig['layout'] = dict()
if 'margin' not in fig['layout']:
    fig['layout']['margin'] = dict(t=50, b=50, r=50, l=50)
fig['layout'].update(autosize=False, width=500, height=500)


plot_url = py.plot(fig, filename='YIGnBu-heatmap', auto_open=False)